### PR TITLE
Revise plotting with matplotlib in parallel

### PR DESCRIPTION
**Pull request by jorgensd**
- Lagrange variants demo should only be executed in real mode
- Skip plotting in parallel for Lagrange variants demo (as they try different to write to the same file).
- TNT demo: Only plot on rank zero as all processes have the same data
- Testing the ubuntu-22.04 runner (https://github.com/actions/virtual-environments/issues/5490)

### CODE REVIEW

File: python/demo/demo_lagrange_variants.py [hunk 1]
```diff
@@ -19,23 +19,29 @@
...
 from petsc4py.PETSc import ScalarType
 
+if np.issubdtype(ScalarType, np.complexfloating):
+    print("Demo should only be executed with DOLFINx real mode")
+    exit(0)
+
 import matplotlib.pylab as plt
+
+import basix
+import basix.ufl_wrapper
```
Comment (mscroggs) [inline python/demo/demo_lagrange_variants.py:37]: These two imports should maybe be after the next comment block that explains them?

File: .github/workflows/pyvista.yml [hunk 2]
```diff
@@ -2,18 +2,18 @@ name: Test visualisation demos
 
 on:
   # Uncomment the below to trigger tests on push
-  # push:
-  #   branches:
-  #     - "**"
+  push:
+    branches:
+      - "**"
```
Comment (mscroggs) [inline .github/workflows/pyvista.yml:7]: Should this be on for every push or just for PRs to main?

**Review by mscroggs — APPROVED** [review]

File: .github/workflows/pyvista.yml (hunk 2 above)
Comment (jorgensd) [reply .github/workflows/pyvista.yml:7]: It is just to be able to check that the CI passes, once it is green I'll revert it

Comment (jorgensd) [inline .github/workflows/pyvista.yml:7]:
```suggestion
  # push:
  #   branches:
  #     - "**"
```

**Review by garth-wells — APPROVED** [review]

**Pull request merged**